src/basicfuns.jl

```diff
@@ -309,9 +309,22 @@ Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and hand
 non-finite values.
 """
 function logaddexp(x::Real, y::Real)
-    # ensure Δ = 0 if x = y = ± Inf
-    Δ = x == y ? zero(x - y) : abs(x - y)
-    max(x, y) + log1pexp(-Δ)
+    if isnan(x) return x end
+    if isnan(y) return y end
+
+    diff = zero(promote_type(typeof(x), typeof(y)))
+    if x < y
+        diff = x - y
+        x = y
+    else
+        diff = y - x
+    end
+
+    if diff >= log(eps())
```

Comment:
This is not correct since the check does not depend on the type of `diff` but the RHS is always a `Float64`. I am not happy about introducing any tolerances here in general, `iszero` is much cleaner.

Reply:
I understand that `diff < log(eps(typeof(diff)))` would be better,  however I don't see how one can use `iszero` in this context. Do you have a precise idea in mind ?

Reply:
Ah, yeah my bad I don't think `iszero` is helpful here. I am surprised that the branch is useful at all. It would be better to use `log1pexp` instead of `log1p(exp(...))`, as in the current implementation, since it is already optimized and avoids the `log1p` and `exp` calls if it is not required for numerical accuracy.